File: Depth First Search(DFS) Algorithm/JavaScript/Depth-First-Search-Algorithm.js
class Graph {
    constructor(vertices) {
        this.vertices = vertices;
        this.adjList = new Array(vertices).fill(null).map(() => []);
        this.visited = new Array(vertices).fill(false);
    }

    addEdge(src, dest) {
        this.adjList[src].push(dest);
        this.adjList[dest].push(src);
    }

    dfs(vertex) {
        this.visited[vertex] = true;
        console.log("Visited", vertex);

        for (const adjVertex of this.adjList[vertex]) {
            if (!this.visited[adjVertex]) {
                this.dfs(adjVertex);
            }
        }
    }
}

const g = new Graph(6);

g.addEdge(0, 1);
g.addEdge(0, 2);
g.addEdge(1, 3);
g.addEdge(1, 4);
g.addEdge(2, 4);
g.addEdge(3, 5);
g.addEdge(4, 5);

g.dfs(0);

File: Depth First Search(DFS) Algorithm/Python/Depth-First-Search-Algorithm.py
class Graph:
    def __init__(self, vertices):
        self.vertices = vertices
        self.adj_list = [[] for _ in range(vertices)]
        self.visited = [False] * vertices

    def add_edge(self, src, dest):
        self.adj_list[src].append(dest)
        self.adj_list[dest].append(src)

    def dfs(self, vertex):
        self.visited[vertex] = True
        print(f"Visited {vertex}")

        for adj_vertex in self.adj_list[vertex]:
            if not self.visited[adj_vertex]:
                self.dfs(adj_vertex)

g = Graph(6)

g.add_edge(0, 1)
g.add_edge(0, 2)
g.add_edge(1, 3)
g.add_edge(1, 4)
g.add_edge(2, 4)
g.add_edge(3, 5)
g.add_edge(4, 5)

g.dfs(0)
